site-src/geps/gep-724.md

```diff
@@ -0,0 +1,624 @@
+type RouteGateways struct {
+  // Allow indicates which Gateways will be allowed to use this route.
+  // Possible values are:
+  // * All: Gateways in any namespace can use this route.
+  // * FromList: Only Gateways specified in GatewayRefs may use this route.
+  // * SameNamespace: Only Gateways in the same namespace may use this route.
+  //
+  // +optional
+  // +kubebuilder:validation:Enum=All;FromList;SameNamespace
+  // +kubebuilder:default=SameNamespace
+  Allow *GatewayAllowType `json:"allow,omitempty"`
+
+  // GatewayRefs must be specified when Allow is set to "FromList". In that
+  // case, only Gateways referenced in this list will be allowed to use this
+  // route. This field is ignored for other values of "Allow".
+  //
+  // +optional
+  GatewayRefs []GatewayReference `json:"gatewayRefs,omitempty"`
+}
+```
+
+#### Added
+To Route Specs:
+```go
+  // ParentRefs references the parent resources this Route should be attached
+  // to. The only kind of parent resource with "Core" support is Gateway. This
+  // API may be extended in the future to support additional kinds of parent
+  // resources such as Routes. It is invalid to reference the same parent
+  // resource more than once.
+  //
+  // +optional
+  // +kubebuilder:validation:MaxItems=16
+  ParentRefs []ParentRef `json:"parentRefs,omitempty"`
+```
+
+And the struct that references:
+```go
+// ParentRef identifies an API object that should be considered a parent of this
+// resource. The only kind of parent resource with "Core" support is Gateway.
+// This API may be extended in the future to support additional kinds of parent
+// resources such as Routes.
+type ParentRef struct {
+  // Group is the group of the referent.
+  //
+  // Support: Core
+  //
+  // +kubebuilder:validation:MinLength=1
+  // +kubebuilder:validation:MaxLength=253
+  // +kubebuilder:default=gateway.networking.k8s.io
+  Group string `json:"group"`
+
+  // Kind is kind of the referent.
+  //
+  // Support: Core (Gateway)
+  // Support: Extended (Other Resources)
+  //
+  // +kubebuilder:validation:MinLength=1
+  // +kubebuilder:validation:MaxLength=253
+  // +kubebuilder:default=Gateway
+  Kind string `json:"kind"`
+
+  // Namespace is the namespace of the referent. When unspecified (empty
+  // string), this will either be:
+  //
+  // * local namespace of the target is a namespace scoped resource
+  // * no namespace (not applicable) if the target is cluster-scoped.
+  //
+  // Support: Extended
+  //
+  // +kubebuilder:validation:MinLength=1
+  // +kubebuilder:validation:MaxLength=253
+  // +optional
+  Namespace string `json:"namespace,omitempty"`
+
+  // Scope represents if this refers to a cluster or namespace scoped resource.
+  // This may be set to "Cluster" or "Namespace".
+  //
+  // Support: Core (Namespace)
+  // Support: Extended (Cluster)
+  //
+	// +kubebuilder:validation:Enum=Cluster;Namespace
+  // +kubebuilder:default=Namespace
+  // +optional
+  Scope string `json:"scope,omitempty"`
```

Comment:
I think you need to add `"";` to the enum or make this a pointer type for it to be optional.

Reply:
Could be left for the implementation, but yes, I agree that this should probably be a pointer, and have the omitted behavior defined to be the same as setting to "Namespace".

Reply:
Updated, thanks!